### ISSUE TITLE
Use safeify instead of eval

### ISSUE THREAD

**Issue by linonetwo** [zadam/trilium#310]
The `execute` in  
https://github.com/zadam/trilium/blob/e02eca87b0d7f3314da2acab7107a9a65f642fd5/src/services/script.js#L64

Uses eval, which can't prevent dead loop and async operation without timeout.

Actually, can use https://github.com/Houfeng/safeify for nodejs and https://github.com/asvd/jailed for browser.

**Comment by zadam** [comment]
This looks interesting. I'm not very concerned about security aspect since Trilium is explicitely single user, but as you mention preventing dead loop and similar things might be very useful since they can currently crash the application.

**Comment by linonetwo** [comment]
I think the code note is a powerful add-on system.

Even Trilium is a PKM tool, people may still share their buttons, cronjobs and aggregation jobs to other people via copy and paste.

If security is under concern, this kind of add-on community may grow.

**Comment by zadam** [comment]
That's true, but then crashing the app is one of the smallest concerns - scripts have total access to note contents (even protected ones while in protected session) and also network access to send it. For the foreseeable future it will be on the user to make sure the script is not harmful.
